### PR TITLE
Training/Inference Module DFP Production

### PR DESCRIPTION
**Pull request by bsuryadevara**
- Added a file to derive DFP command line arguments.
- Created partial (some stages are implemented in cpp) inference pipeline module
- Created training pipeline module
- Added pipeline executable files to run the inference/training pipeline modules

### CODE REVIEW

File: examples/digital_fingerprinting/production/morpheus/dfp_duo_modules_inference.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        LinearModulesStage(config,
+                           module_config,
+                           input_port_name="input",
+                           output_port_name="output",
+                           output_type=MultiMessage))
+
+    pipeline.add_stage(MonitorStage(config, description="Preprocessing & Inference rate", smoothing=0.001))
+
+    pipeline.add_stage(
+        FilterDetectionsStage(config, threshold=2.0, filter_source=FilterSource.DATAFRAME, field_name='mean_abs_z'))
```
Comment (drobison00) [inline examples/digital_fingerprinting/production/morpheus/dfp_duo_modules_inference.py:287]: Can we roll this into the inference module, or is there some reason to keep it outside?

File: examples/digital_fingerprinting/production/morpheus/dfp_duo_modules_inference.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    pipeline.add_stage(
+        LinearModulesStage(config,
+                           post_proc_config,
+                           input_port_name="input",
+                           output_port_name="output",
+                           input_type=MultiMessage,
+                           output_type=MultiMessage))
+
+    # Exclude the columns we don't want in our output
+    pipeline.add_stage(SerializeStage(config, exclude=['batch_count', 'origin_hash', '_row_hash', '_batch_id']))
```
Comment (drobison00) [inline examples/digital_fingerprinting/production/morpheus/dfp_duo_modules_inference.py:298]: Same here.

File: examples/digital_fingerprinting/production/morpheus/dfp_duo_modules_inference.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+    # Write all anomalies to a CSV file
+    pipeline.add_stage(WriteToFileStage(config, filename="dfp_detections_duo.csv", overwrite=True))
```
Comment (drobison00) [inline examples/digital_fingerprinting/production/morpheus/dfp_duo_modules_inference.py:301]: And here.

File: examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+        LinearModulesStage(config,
+                           module_config,
+                           input_port_name="input",
+                           output_port_name="output",
+                           output_type=MultiMessage))
+
+    pipeline.add_stage(MonitorStage(config, description="Preprocessing & Inference rate", smoothing=0.001))
+
+    pipeline.add_stage(
+        FilterDetectionsStage(config, threshold=2.0, filter_source=FilterSource.DATAFRAME, field_name='mean_abs_z'))
```
Comment (drobison00) [inline examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py:289]: Can this go into the module?

File: examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    pipeline.add_stage(
+        LinearModulesStage(config,
+                           post_proc_config,
+                           input_port_name="input",
+                           output_port_name="output",
+                           input_type=MultiMessage,
+                           output_type=MultiMessage))
+
+    # Exclude the columns we don't want in our output
+    pipeline.add_stage(SerializeStage(config, exclude=['batch_count', 'origin_hash', '_row_hash', '_batch_id']))
```
Comment (drobison00) [inline examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py:300]: Same here.

File: examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+    # Write all anomalies to a CSV file
+    pipeline.add_stage(WriteToFileStage(config, filename="dfp_detections_azure.csv", overwrite=True))
```
Comment (drobison00) [inline examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py:303]: Same here.

File: examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+                    tracking_uri=kwargs["tracking_uri"],
+                    source="azure")
+
+    da.init()
+
+    config: Config = get_ae_config(labels_file="data/columns_ae_azure.txt",
+                                   userid_column_name="username",
+                                   timestamp_column_name="timestamp")
+
+    # Specify the column names to ensure all data is uniform
```
Comment (drobison00) [inline examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py:140]: Can you pull source_column_info, preprocess_column_info, and the module config in another file, just so its easier to digest the workflow quickly, and then if someone wants/needs to they can dig into the config.

**Review by drobison00 — CHANGES_REQUESTED** [review]
Added a couple comments; for the most part everything looks good. Just want to polish the presentation a bit.

File: examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py (hunk 7 above)
Comment (bsuryadevara) [reply examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_inference.py]: Done.

**Review by drobison00 — APPROVED** [review]
Looks good, just change encoding from 'latin1' to utf-8, and double check pipelines run correctly.

**Pull request merged**